src/src/Globals/Nodes.h

```diff
@@ -1,10 +1,11 @@
 #ifndef GLOBALS_NODES_H
 #define GLOBALS_NODES_H
 
+#ifdef FEATURE_ESP_P2P
```

Comment:
You're at least missing an include here.

`#include "../../ESPEasy_common.h"`

Everywhere you're using a check for this define, you need to have included the ESPEasy_common.h file.
So in any .cpp or .h file, it must be included in one of the .h files you include or you specifically include it yourself.

Reply:
but then why builds correctly locally ?

Reply:
> but then why builds correctly locally ?

That's a very good question, as it seems to depend on the platform which is used to build (Windows vs. Linux) and it shouldn't.
Windows also doesn't seem to be as picky on the path of the include, but it should.

Believe me, I've been bitten by this myself quite often.

Reply:
updated that header file and checked and seems all the others are fine.

Reply:
that did the trick. Thank you for your help.